### PR TITLE
Adding importDetail when updating existing publication by import entry

### PR DESCRIPTION
**Pull request by truhacevkir**

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 1]
```diff
@@ -179,6 +179,13 @@ public Publication createPublicationFromImportedEntry(Publication inputData, Imp
         return insertResource(newResource);
     }
 
+    public Publication updatePublicationByImportEntry(Publication publication, ImportSource importSource) {
+        if (nonNull(importSource)) {
+            publication.addImportDetail(new ImportDetail(Instant.now(), importSource));
+        }
+        return updateResourceService.updatePublicationButDoNotChangeStatus(publication);
```
Comment (truhacevkir) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:186]: Do we want the same timestamp as in modifiedDate?

File: cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java [hunk 2]
```diff
@@ -181,7 +181,8 @@ private Publication update(PublicationRepresentations publicationRepresentations
 
     private PublicationRepresentations persistUpdatedPublication(
         PublicationRepresentations publicationRepresentations) {
-        resourceService.updatePublication(publicationRepresentations.getExistingPublication());
+        resourceService.updatePublicationByImportEntry(publicationRepresentations.getExistingPublication(),
+                                                       ImportSource.fromSource(Source.CRISTIN));
```
Comment (truhacevkir) [inline cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java:185]: Adding importDetail in cristin-import

File: brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java [hunk 3]
```diff
@@ -265,8 +265,11 @@ private UriWrapper discardedFilesReportUri(Publication publication, S3Event s3Ev
     }
 
     private BrageMergingReport persistInDatabaseAndCreateMergeReport(Publication publicationForUpdate,
-                                                                     Publication existinPublication) {
-        var newImage = resourceService.updatePublication(publicationForUpdate);
+                                                                     Publication existinPublication,
+                                                                     PublicationRepresentation publicationRepresentation) {
+        var customerName = publicationRepresentation.brageRecord().getCustomer().getName();
+        var importSource = ImportSource.fromBrageArchive(customerName);
+        var newImage = resourceService.updatePublicationByImportEntry(publicationForUpdate, importSource);
```
Comment (truhacevkir) [inline brage-import/src/main/java/no/sikt/nva/brage/migration/lambda/BrageEntryEventConsumer.java:272]: adding import detail in brage-import

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java (hunk 1 above)
Comment (joachimjorgensen) [reply publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:186]: I guess that would be best, if it's not too much hassle

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 4]
```diff
@@ -179,6 +179,13 @@ public Publication createPublicationFromImportedEntry(Publication inputData, Imp
         return insertResource(newResource);
     }
 
+    public Publication updatePublicationByImportEntry(Publication publication, ImportSource importSource) {
+        if (nonNull(importSource)) {
+            publication.addImportDetail(new ImportDetail(Instant.now(), importSource));
+        }
```
Comment (joachimjorgensen) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:185]: Throw exception if null instead? Doesn't really make sense to use this method if `importSource` is null

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java [hunk 5]
```diff
@@ -1187,6 +1187,18 @@ void shouldThrowBadRequestWhenAttemptingToDeletePublishedPublication() throws Ap
...
     }
 
+    @Test
+    void shouldUpdatePublicationByImportedEntry() throws ApiGatewayException {
+        var publication = createPublishedResource();
+        publication.setDoi(randomDoi());
+        var updatedPublication = resourceService.updatePublicationByImportEntry(publication,
+                                                                   ImportSource.fromSource(Source.CRISTIN));
+
+        assertThat(updatedPublication.getImportDetails().getFirst().importSource().getSource(),
```
Comment (joachimjorgensen) [inline publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:1197]: Add an assertion that `getImportDetails().size == 1`? To make sure we're not call the method too many times

Comment (joachimjorgensen) [reply publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:1197]: Or maybe it's better to assert that in the handler tests

**Review by joachimjorgensen — APPROVED** [review]

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java (hunk 5 above)
Comment (truhacevkir) [reply publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:1197]: Will assert in both of tests

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java (hunk 4 above)
Comment (truhacevkir) [reply publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:185]: throwing IllegalArgumentException when passing null to this method

**Review by joachimjorgensen — APPROVED** [review]

**Pull request merged**